modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/exportimport/portlet/preferences/processor/DLExportImportPortletPreferencesProcessor.java

```diff
@@ -303,49 +330,48 @@ public PortletPreferences processImportPortletPreferences(
 
 		if (stagingPreferencesMappingJSONObject != null) {
 			try {
-				long folderRepositoryId =
-					stagingPreferencesMappingJSONObject.getLong(
-						"folderRepositoryId");
-				String folderUuid =
-					stagingPreferencesMappingJSONObject.getString("folderUuid");
-
-				long folderId = DLFolderConstants.DEFAULT_PARENT_FOLDER_ID;
-
-				if (Validator.isNotNull(folderUuid)) {
-					DLFolder dlFolder =
-						_dlFolderLocalService.getDLFolderByUuidAndGroupId(
-							folderUuid, folderRepositoryId);
-
-					folderId = dlFolder.getFolderId();
-				}
+				portletPreferences.setValue(
+					_PREFERENCE_KEY_ROOT_FOLDER_EXTERNAL_REFERENCE_CODE,
+					stagingPreferencesMappingJSONObject.getString(
+						_PREFERENCE_KEY_ROOT_FOLDER_EXTERNAL_REFERENCE_CODE));
 
 				portletPreferences.setValue(
-					"rootFolderId", String.valueOf(folderId));
+					_PREFERENCE_KEY_SELECTED_GROUP_EXTERNAL_REFERENCE_CODE,
+					stagingPreferencesMappingJSONObject.getString(
+						_PREFERENCE_KEY_SELECTED_GROUP_EXTERNAL_REFERENCE_CODE));
+
 				portletPreferences.setValue(
-					"selectedRepositoryId", String.valueOf(folderRepositoryId));
+					_PREFERENCE_KEY_SELECTED_REPOSITORY_EXTERNAL_REFERENCE_CODE,
+					stagingPreferencesMappingJSONObject.getString(
+						_PREFERENCE_KEY_SELECTED_REPOSITORY_EXTERNAL_REFERENCE_CODE));
 
 				return portletPreferences;
 			}
-			catch (PortalException | ReadOnlyException exception) {
-				throw new PortletDataException(exception);
+			catch (ReadOnlyException readOnlyException) {
+				throw new PortletDataException(readOnlyException);
 			}
 		}
 
-		// Root folder ID is set, only import that
+		// Root folder ERC is set, only import that
 
-		long rootFolderId = GetterUtil.getLong(
-			portletPreferences.getValue("rootFolderId", null));
+		String rootFolderExternalReferenceCode = portletPreferences.getValue(
+			_PREFERENCE_KEY_ROOT_FOLDER_EXTERNAL_REFERENCE_CODE, null);
 
-		if (rootFolderId != DLFolderConstants.DEFAULT_PARENT_FOLDER_ID) {
+		if (!Validator.isBlank(rootFolderExternalReferenceCode)) {
 			Element foldersElement =
 				portletDataContext.getImportDataGroupElement(DLFolder.class);
 
 			List<Element> folderElements = foldersElement.elements();
 
 			if (!folderElements.isEmpty()) {
 				try {
+					Element folderElement = folderElements.get(0);
+
+					long rootFolderId = _getFolderId(
+						folderElement, portletDataContext);
```

Comment:
Can you move this closer to the first usage?

Reply:
See change here: [cc09b54](https://github.com/liferay-content-management/liferay-portal/pull/6244/commits/cc09b548151fa6f38670c34e282c6619a87cf855)